pkg/util/httputil/client.go

```diff
@@ -25,6 +25,12 @@ var DefaultClient = NewClientWithTimeout(StandardHTTPTimeout)
 // StandardHTTPTimeout is the default timeout to use for HTTP connections.
 const StandardHTTPTimeout time.Duration = 3 * time.Second
 
+// RequestHeaders are the headers to be part of the request
+type RequestHeaders struct {
```

Comment:
Idle thought: the Go stdlib already has a `Header` type[^1] that could also have been used here.

[^1]: https://pkg.go.dev/net/http#Header